docs/.vuepress/plugins/sidebar/SidebarItem.vue

```diff
@@ -65,94 +59,86 @@ export default {
-      let newName = this.newItem.name;
+    getPathName() {
       const newPath = `${this.folderName}/${
         this.newItem.path.split(`/${this.folderName}/`)[1]
       }`;
-      this.pagesNew.forEach((page) => {
-        if (
-          String(page.relativePath.replace("/README.md", "")) ===
-          String(newPath)
-        ) {
-          newName = page.title;
-        }
-      });
-
+      let newName = newPath.substr(newPath.lastIndexOf('/'));
+      newName = newName.replace('.md', '');
       return newName;
     },
-    link() {
+    getLink() {
       let hasReadme = false;
       let link = `/${this.folderName}/${
         this.newItem.path.split(`/${this.folderName}/`)[1]
       }`;
 
-      console.log(link);
-
       this.newItem.children &&
         this.newItem.children.length > 0 &&
-        this.newItem.children.forEach((child) => {
-          if (child.name === "README.md") {
+        this.newItem.children.forEach(child => {
+          if (child.name === 'README.md') {
             hasReadme = true;
           }
         });
 
-      if (link.includes(".md") || hasReadme) {
-        return link.replace("README", "").replace(".md", "");
+      if (link.includes('.md') || hasReadme) {
+        return link.replace('README', '').replace('.md', '');
       }
 
-      return "";
+      return '';
     },
   },
   methods: {
-    toggle() {
+    toggleFolder() {
+      console.log(this.pages);
       if (this.isFolder) {
         this.isOpen = !this.isOpen;
       }
     },
     containsURL() {
-      return this.href !== ""
-        ? this.href.includes("/" + this.newItem.name)
+      return this.href !== ''
+        ? this.href.includes('/' + this.newItem.name)
         : false;
     },
-    isItemSelected() {
-      if (this.href === "") return false;
-
-      const newHref =
-        this.href[this.href.length - 1] === "/"
-          ? this.href.substring(0, this.href.length - 1)
-          : this.href;
-
-      const windowLocation = newHref.includes(this.folderName)
-        ? `/${this.folderName}/${newHref
-            .split(`/${this.folderName}/`)[1]
-            .replace(".html", "")}`
-        : this.href;
-
-      return this.link === windowLocation;
-    },
-  },
-  mounted() {
-    this.href = window.location.href;
-    const isURLItem = this.href.includes("/" + this.newItem.name);
-
-    if (this.newItem.name === this.folderName || isURLItem) {
-      this.isOpen = true;
-    }
+    // isItemSelected() {
+    //   if (this.href === '') return false;
+
+    //   const newHref =
+    //     this.href[this.href.length - 1] === '/'
+    //       ? this.href.substring(0, this.href.length - 1)
+    //       : this.href;
+
+    //   const windowLocation = newHref.includes(this.folderName)
+    //     ? `/${this.folderName}/${newHref
+    //         .split(`/${this.folderName}/`)[1]
+    //         .replace('.html', '')}`
+    //     : this.href;
+
+    //   return this.link === windowLocation;
+    // },
```

Comment:
Is this comment necessary?

Reply:
For now yes, as I mentioned in the `changes`, the sidebar is still in wip status and still experimenting with it.